### PR TITLE
Add basic search functionality

### PR DESCRIPTION
**Pull request by Neeraj-Ghodla**
## Description
Basic search functionality is implemented by creating an index-data.json file containing all the headings, descriptions, and anchor links at build time.

We then use the flexsearch library to build an index using the data from index-data.json to provide local search functionality.

## How to test
* Either press `/` or click on the search bar to open it
* Type a keyword to search. If the keyword exists in any heading or description, you should see some search results otherwise you'll see the "Nothing Found" text.

Desktop View

Mobile View

### CODE REVIEW

**Comment by IsaacMurtagh** [comment]
Can you add a screenshot an fix pipeline?

File: astro.config.mjs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { defineConfig } from 'astro/config';
+import flexsearchPlugin from './src/plugins/flexsearch';
```
Comment (IsaacMurtagh) [inline astro.config.mjs:2]: I thought this doesn't work, is this why the build is broken?

**Comment by tom-blake** [comment]
@Neeraj-Ghodla Can you please drop some mobile/desktop screenshots and a test plan into this PR plz? 🙏

File: astro.config.mjs (hunk 1 above)
Comment (Neeraj-Ghodla) [reply astro.config.mjs:2]: It wasn't because of this. I've fixed the build.

**Review by IsaacMurtagh — APPROVED** [review]
Nice, I like it

**Pull request merged**